hbase-server/src/main/java/org/apache/hadoop/hbase/coordination/ZKSplitTransactionCoordination.java

```diff
@@ -163,6 +166,10 @@ public void waitForSplitTransaction(final RegionServerServices services, Region
         }
         Thread.sleep(100);
         spins++;
+        if (spins > SPLIT_WAIT_TIMEOUT) {
```

Comment:
@apurtell we retry here for ~20-24s. We try to update ZNode to RS_ZK_REQUEST_REGION_SPLIT [here](https://github.com/apache/hbase/pull/809/files#diff-fdc12d2818ed13d9c47380d943712d9fR164) (L:164) to request master for split. If it fails due to BVE, well we again come back there and re-do the same thing until we succeed(all this code inside while loop). The success is measured by RS getting positive response from master in the form of ZNode state being updated to RS_ZK_REGION_SPLITTING [here](https://github.com/apache/hbase/pull/809/files#diff-fdc12d2818ed13d9c47380d943712d9fR179)(L:179) from where RS exits from this while loop successfully, if not successful for ~20-24s, on this line(L:169) we are introducing exit condition with IOE, which will be responsible to rollback splitting after catching this IOE [here](https://github.com/apache/hbase/blob/f7ca22fd4890a952fcf8fa66254fe08567bf2306/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/SplitRequest.java#L85)

Reply:
Here, we are already in a while loop, and the only way to come out of it with successful splitting is to get `RS_ZK_REGION_SPLITTING` from ZNode which should have been done by master, which would have happened only if RS had successfully transitioned ZNode to `RS_ZK_REQUEST_REGION_SPLIT`, which would happen only if BVE is not encountered. Until then, we are anyways in this while loop 
`while (!(coordinationManager.getServer().isStopped() || services.isStopping())) {` and retrying `transitionSplittingNode()`